app/src/main/java/dominando/android/livros/BookFormFragment.kt

```diff
@@ -33,78 +31,53 @@ class BookFormFragment : BaseFragment() {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        binding = DataBindingUtil.inflate(
-                inflater, R.layout.fragment_book_form, container, false
-        )
-        return binding.root
+    ): View {
+        val binding = DataBindingUtil.inflate(
+                inflater,
+                R.layout.fragment_book_form,
+                container, false) as FragmentBookFormBinding
+
+        return binding.run {
+            lifecycleOwner = this@BookFormFragment
+            content.presenter = this@BookFormFragment
+            content.viewModel = this@BookFormFragment.viewModel
+            root
+        }
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        if (viewModel.book == null) {
-            viewModel.book = arguments?.getParcelable("book") ?: Book()
+        arguments?.getParcelable<Book>("book")?.let {
```

Comment:
Use `EXTRA_BOOK` constant here. (see comment above)

Reply:
The same solution above